### PR TITLE
(doc): Document RevealMD rendering

### LINKED ISSUE #300: Pelican reveal-md
Do we want to enable 'by default' <https://github.com/brookskindle/pelican-revealmd> on blog-o-matic and documentation website so that we can upload presentations of Elegant directly in website?

### PR DESCRIPTION
**Pull request by iranzo**
Signed-off-by: Pablo Iranzo Gómez <Pablo.Iranzo@gmail.com>

## Prerequisites

- [x] My Pull Request is filled against the `next` branch
- [ ] All commits in my Pull Request conform to [Elegant Git commit Guidelines](https://elegant.oncrashreboot.com/git-commit-guidelines)

## Recommended Steps

- [x] My patch adds a new feature, therefore I have also added a help article about it
- [ ] My patch changes Elegant behavior, therefore I have updated the help article to reflect this change
- [x] My commits are [signed](https://help.github.com/en/articles/signing-commits)

## Description
Closes #300

### CODE REVIEW

File: documentation/content/Extra Customization/revealjs.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+### Issues
+
+You can find more details on the [initial issues found](https://iranzo.github.io/blog/2019/01/20/pelican-revealjs/)
+and how they were fixed.
+
+## Usage
+
+- Enable pelican-revealmd plugin from this repository: <https://github.com/iranzo/pelican-revealmd>
+- Create your revealMD files and name them as 'revealjs'
```
Comment (talha131) [inline documentation/content/Extra Customization/revealjs.md]: Need to make it clear that files are Markdown and their extension is `revealjs`. 

An example file name would be nice `my_presentation.revealjs`.

File: documentation/content/Extra Customization/revealjs.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+author: Pablo Iranzo Gómez
+title: Pelican-Elegant and RevealMD presentations
+tags: pelican, foss, blog, python, github, blog-o-matic, linux, pandoc
+layout: post
```
Comment (talha131) [inline documentation/content/Extra Customization/revealjs.md]: What is this variable for? `layout`.

File: documentation/content/Extra Customization/revealjs.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+date: 2019-07-22 22:12:17 +0100
+comments: true
+category: tech
```
Comment (talha131) [inline documentation/content/Extra Customization/revealjs.md]: Change it to `Supported Plugins`.

**Comment by iranzo** [comment]
> @iranzo is that it? Don't we need #273 for it to work?

It would be better to have 273, as generated 'pages' will have all meta tags, etc that will make indexing by browsers better, but it's not a priority to 'make it work'

File: documentation/content/Extra Customization/revealjs.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+description:
+---
```
Comment (talha131) [inline documentation/content/Extra Customization/revealjs.md]: Please add slug variable. like `reveal-md-presentations-in-elegant`.

File: documentation/content/Extra Customization/revealjs.md (hunk 2 above)
Comment (iranzo) [reply documentation/content/Extra Customization/revealjs.md]: It's in the template I use for my blog, probably from jekyll time :) (removed)

File: documentation/content/Extra Customization/revealjs.md (hunk 3 above)
Comment (iranzo) [reply documentation/content/Extra Customization/revealjs.md]: Done in next commit

File: documentation/content/Extra Customization/revealjs.md (hunk 4 above)
Comment (iranzo) [reply documentation/content/Extra Customization/revealjs.md]: done in next commit

File: documentation/content/Extra Customization/revealjs.md (hunk 1 above)
Comment (iranzo) [reply documentation/content/Extra Customization/revealjs.md]: In next commit

**Comment by talha131** [comment]
@iranzo last thing left is to place this article in "Supported Plugins" folder.

**Comment by jackdewinter** [comment]
Would you mind if I worked with you to rework the article to keep the information but make it consistent with the other articles I am working on?

**Comment by iranzo** [comment]
> Would you mind if I worked with you to rework the article to keep the information but make it consistent with the other articles I am working on?

No, go ahead!

**Comment by talha131** [comment]
@iranzo I looked into it and realized it can turned into a powerful feature. For example, we can let user specify the revealmd theme to use.

Adding a blank template is necessary. 

We can build a more power feature on top of blank template. Like letting user use different layouts or themes for his posts. So one post uses the default article layout and colors. Other post may have a totally different layout and colors.

We need to look more deeply into it. I will merge this PR when we have a design finalized.

**Comment by jackdewinter** [comment]
Working with Pablo to understand the settings and such, to make sure I can properly document it.

**Comment by iranzo** [comment]
@talha I can't see what's wrong with this 'PR' based on the travis builds, errors seem to come from another file not touched by this one.

Thanks

**Comment by talha131** [comment]
> @talha131 I can't see what's wrong with this 'PR' based on the travis builds, errors seem to come from another file not touched by this one.

Yeah. These errors are due to utterances comment system.

**Comment by iranzo** [comment]
> > @talha131 I can't see what's wrong with this 'PR' based on the travis builds, errors seem to come from another file not touched by this one.
> 
> Yeah. These errors are due to utterances comment system.

Rebased on top of current next as I tought it was fixed, but still failing :-)

Thanks!

**Comment by talha131** [comment]
Another reason for failure is  a SVG icon. Need to dig deeper to fix it.

**Comment by iranzo** [comment]
@talha131 now passing CI :-)

**Comment by talha131** [comment]
> @talha131 now passing CI :-)

yeah, it got fixed here b05b35e8

**Comment by iranzo** [comment]
Got this rebased on latest next to get this documentation 'in'

**Comment by iranzo** [comment]
I've rebased again to get this merged

**Pull request merged**